benchmarks/README.md

```diff
@@ -0,0 +1,100 @@
+# Mesa Performance Benchmarks
+
+The `/benchmarks` directory contains tools for benchmarking the Mesa library performance in order to compare improvements and regressions  in efficiency across different modifications of Mesa.
+
+MESA uses several example base models for benchmarking performance (BoltzmannWealth, Schelling, BoidFlockers, and WolfSheep) by calculating the initialization time and run time for each of these models. These example models can be found in the `/mesa/examples` directory.
+
+## Available Files
+
+- `configurations.py`: Contains model configurations for benchmarking
+- `global_benchmark.py`: Main script for running benchmarks
+- `compare_timings.py`: Tool to compare results between benchmark runs
+
+## How to Use
+
+
+### 1. Benchmark Configuration
+
+The `configurations.py` file defines which models to benchmark and their parameters. Each model has:
+
+- `small` and `large` configurations for testing different scales
+- Parameters for:
+  - `seeds`: Number of different random seeds to use
+  - `replications`: Number of times to repeat the benchmark for each seed
+  - `steps`: Number of model steps to run
+  - `parameters`: Model-specific parameters
+
+### 2. Running Benchmarks
+
+To run the benchmarks:
+
+```bash
+pip install tabulate # if you don't have it yet
+python global_benchmark.py
+```
+
+This will:
+- Run all models defined in `configurations.py` with their respective configurations
+- Measure initialization time and run time for each model
+- Save results to a pickle file named `timings_X.pickle` (where X is an incremental number)
+- Display a summary of results in the terminal
+
+
+>**Noteworthy :** the pickle file created by the benchmark is not under git control. So you can run the benchmark on the master branch first, switch to your development branch, change the file name for the results, and run the benchmarks again.
```

Comment:
really minor detail: before you said that the number auto increments (which is correct if I recall), so then you would not have to change the file name as stated here (by me in an earlier comment).